indico/web/client/js/jquery/widgets/track_role_widget.js

```diff
@@ -5,86 +5,41 @@
-      var roleData = self.roleData[track][role];
-
-      $field.val(JSON.stringify(_.map(roleData, _.propertyOf(self.options.userData))));
-      $field.principalfield({
-        allowExternalUsers: false,
-        multiChoice: true,
-        overwriteChoice: false,
-
-        render: function(people) {
-          var $placeholder = $container.find('.placeholder');
-          var $list = $container.find('.user-role-list').empty();
-
-          $placeholder.toggle(!people.length);
-          _.sortBy(people, _.property('name')).forEach(function(person) {
-            $list.append(
-              $('<li class="event-user">')
-                .text(person.name)
-                .append('<a class="remove-user icon-cross hide-if-locked">')
-                .attr('data-user-id', +person.id)
-            );
-          });
-        },
-
-        onAdd: function(people) {
-          people.forEach(function(person) {
-            // XXX: person.id is a string for users from search results
-            roleData.push(+person.id);
-            self._update();
-          });
-        },
-      });
-
-      $container.on('click', '.remove-user', function(evt) {
-        evt.preventDefault();
-        var $userElement = $(this).closest('li');
-        var userId = $userElement.data('userId');
-        $userElement.remove();
-        roleData = self.roleData[track][role] = _.without(roleData, userId);
-        self._update();
-        $field.principalfield('removeOne', userId);
-      });
-
-      $addUserButton.on('click', function(evt) {
-        evt.preventDefault();
-        $field.principalfield('choose');
-      });
+      permissionsInfo: null,
     },
 
-    _update: function() {
-      this.element.val(JSON.stringify(this.roleData));
+    _updateValue: function(newValue, trackId) {
+      const roleData =  JSON.parse(this.element.val())
+      roleData[trackId] = newValue;
+      this.element.val(JSON.stringify(roleData));
     },
 
-    _setUp: function($trackConfig) {
-      var self = this;
-      $trackConfig.find('.js-role-container').each(function() {
-        self._setUpRoleContainer($(this));
-      });
+    _renderACLField: function(trackId, value) {
+      const permissionInfo = this.options.permissionsInfo;
+      const onChange = (newValue) => this._updateValue(newValue, trackId);
+      const element = document.querySelector('#track-roles-' + trackId);
+      const component = React.createElement(
+        TrackACLField, {value, permissionInfo, onChange}
+      );
+      ReactDOM.render(component, element);
     },
 
     _create: function() {
-      var self = this;
-      var $container = self.options.containerElement;
-      self.roleData = JSON.parse(self.element.val());
-      $container.find('.js-track-config').each(function() {
-        self._setUp($(this));
+      const roleData = JSON.parse(this.element.val());
+      Object.entries(roleData).map(([trackId, value]) => {
```

Comment:
Don't use `.map()` if you don't need the return value; use `.forEach()` instaed.